openmdao/core/total_jac.py

```diff
@@ -271,15 +271,24 @@ def __init__(self, problem, of, wrt, use_abs_names, return_format, approx=False,
                 self.simul_coloring = driver._coloring_info['coloring']
 
                 # if we don't get wrt and of from driver, turn off coloring
-                if self.simul_coloring is not None and \
-                   (prom_wrt != driver_wrt or prom_of != driver_of):
-                    msg = ("compute_totals called using a different list of design vars and/or "
-                           "responses than those used to define coloring, so coloring will "
-                           "be turned off.\ncoloring design vars: %s, current design vars: "
-                           "%s\ncoloring responses: %s, current responses: %s." %
-                           (driver_wrt, wrt, driver_of, of))
-                    issue_warning(msg, category=DerivativesWarning)
-                    self.simul_coloring = None
+                if self.simul_coloring is not None:
+                    ok = True
+                    if prom_wrt != driver_wrt:
+                        ok = False
+                    # TODO: there's some weirdness here where sometimes the
+                    # of vars are absolute and sometimes they're not...
+                    for pof, aof, dof in zip(prom_of, of, driver_of):
```

Comment:
This loop could be placed under an else: branch. When prom_wrt is not equal to driver_wrt, then ok is already false, so no need to check the ofs.

Reply:
placed loop under else branch